File: Client/app/src/main/java/ua/nure/havrysh/kindergarten/activities/MarkActivity.java
package ua.nure.havrysh.kindergarten.activities;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;

import java.sql.Date;

import kindergarten.hakito.kindergartenclient.R;
import kindergarten.hakito.kindergartenclient.databinding.ActivityMarkBinding;
import retrofit2.Call;
import retrofit2.Response;
import ua.nure.havrysh.kindergarten.model.Mark;
import ua.nure.havrysh.kindergarten.rest.MyCallback;
import ua.nure.havrysh.kindergarten.rest.Rest;

public class MarkActivity extends BaseEditingActivity {

    private static final String EXTRA_MARK_ID = "mark_id", EXTRA_DATE = "date";

    public static void startForCreating(Context context, long date) {
        Intent intent = new Intent(context, MarkActivity.class);
        intent.putExtra(EXTRA_DATE, date);
        context.startActivity(intent);
    }

    public static void start(Context context, long markId) {
        Intent intent = new Intent(context, MarkActivity.class);
        intent.putExtra(EXTRA_MARK_ID, markId);
        context.startActivity(intent);

    }

    ActivityMarkBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_mark);

        long markId = getIntent().getLongExtra(EXTRA_MARK_ID, 0);
        if (markId > 0) {
            Rest.get().getMark(markId).enqueue(new MyCallback<Mark>() {
                @Override
                public void onResponse(Call<Mark> call, Response<Mark> response) {
                    Mark m = response.body();
                    binding.setMark(m);
                }

                @Override
                public void onFailure(Call<Mark> call, Throwable t) {

                }
            });
        } else {
            long date = getIntent().getLongExtra(EXTRA_DATE, 0);
            Mark mark = new Mark(new Date(date));

            binding.setMark(mark);
        }

    }
}

File: Client/app/src/main/java/ua/nure/havrysh/kindergarten/activities/MarkActivity.kt
package ua.nure.havrysh.kindergarten.activities

import android.content.Context
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import kindergarten.hakito.kindergartenclient.R
import kindergarten.hakito.kindergartenclient.databinding.ActivityMarkBinding
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import ua.nure.havrysh.kindergarten.App
import ua.nure.havrysh.kindergarten.model.Mark
import ua.nure.havrysh.kindergarten.rest.Rest
import java.sql.Date

class MarkActivity : BaseEditingActivity() {
    
    lateinit var binding: ActivityMarkBinding
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_mark)
        
        val markId = intent.getStringExtra(EXTRA_MARK_ID)
        if (markId?.isNotEmpty() ?: false) {
            loadMark(markId)
        } else {
            var date = intent.getLongExtra(EXTRA_DATE, 0)
            val mark = Mark(date = Date(date))
            
            binding.mark = mark
        }
    }
    
    private fun loadMark(markId: String) {
        App.scope.launch(Dispatchers.IO) {
            val mark = Rest.get().getMark(markId).await()
            withContext(Dispatchers.Main) { binding.mark = mark }
        }
    }
    
    companion object {
        
        private val EXTRA_MARK_ID = "mark_id"
        private val EXTRA_DATE = "date"
        
        fun startForCreating(context: Context, date: Long) {
            val intent = Intent(context, MarkActivity::class.java)
            intent.putExtra(EXTRA_DATE, date)
            context.startActivity(intent)
        }
        
        fun start(context: Context, markId: String) {
            val intent = Intent(context, MarkActivity::class.java)
            intent.putExtra(EXTRA_MARK_ID, markId)
            context.startActivity(intent)
        }
    }
}
